lib/pickle.js

```diff
@@ -212,7 +215,9 @@ var Pickle = (function () {
 
   Pickle.prototype.resize = function (newCapacity) {
     newCapacity = alignInt(newCapacity, PAYLOAD_UNIT)
-    this.header = Buffer.concat([this.header, new Buffer(newCapacity)])
+    var newBuffer = typeof Buffer.allocUnsafe === 'function'
+      ? Buffer.allocUnsafe(newCapacity) : new Buffer(+newCapacity)
+    this.header = Buffer.concat([this.header, newBuffer])
```

Comment:
Maybe make a non-exported function:

```js
var bufferAllocUnsafe = function (size) {
  return (typeof Buffer.allocUnsafe === 'function')
    ? Buffer.allocUnsafe(size)
    : new Buffer(size)
}
```

And then this code could read

```js
this.header = Buffer.concat([this.header, bufferAllocUnsafe(+newCapacity)])
```

And the other cases above could also use it, e.g.

```js
this.header = bufferAllocUnsafe(0)
```